src/plugins/shared_dictionaries_plugin/shared_dictionaries_column_processor.cpp

```diff
@@ -0,0 +1,291 @@
+    }
+  }
+}
+
+template <typename T>
+std::shared_ptr<const BaseCompressedVector> SharedDictionariesColumnProcessor<T>::_create_new_attribute_vector(
+    const std::shared_ptr<DictionarySegment<T>> segment, const std::shared_ptr<const pmr_vector<T>> shared_dictionary,
+    const PolymorphicAllocator<T>& allocator) {
+  const auto chunk_size = segment->size();
+  const auto max_value_id = static_cast<uint32_t>(shared_dictionary->size());
+
+  auto uncompressed_attribute_vector = pmr_vector<uint32_t>{allocator};
+  uncompressed_attribute_vector.reserve(chunk_size);
+
+  for (auto chunk_index = ChunkOffset{0}; chunk_index < chunk_size; ++chunk_index) {
+    const auto search_value_opt = segment->get_typed_value(chunk_index);
+    if (search_value_opt) {
+      // Find and add new value id using binary search
+      const auto search_iter =
+          std::lower_bound(shared_dictionary->cbegin(), shared_dictionary->cend(), search_value_opt.value());
+      Assert(search_iter != shared_dictionary->end(), "Shared dictionary does not contain value.");
+      const auto found_index = std::distance(shared_dictionary->cbegin(), search_iter);
+      uncompressed_attribute_vector.emplace_back(found_index);
+    } else {
+      // Assume that search value is NULL
+      uncompressed_attribute_vector.emplace_back(max_value_id);
+    }
+  }
+
+  return std::shared_ptr<const BaseCompressedVector>(compress_vector(
+      uncompressed_attribute_vector, VectorCompressionType::FixedWidthInteger, allocator, {max_value_id}));
+}
+
+// Copied from DictionarySegment::memory_usage
+template <typename T>
+size_t SharedDictionariesColumnProcessor<T>::_calc_dictionary_memory_usage(
+    const std::shared_ptr<const pmr_vector<T>> dictionary) {
+  if constexpr (std::is_same_v<T, pmr_string>) {
+    return string_vector_memory_usage(*dictionary, MemoryUsageCalculationMode::Full);
+  }
+  return dictionary->size() * sizeof(typename decltype(dictionary)::element_type::value_type);
+}
+
+template <typename T>
+bool SharedDictionariesColumnProcessor<T>::_should_merge(
+    const double jaccard_index, const size_t current_dictionary_size, const size_t shared_dictionary_size,
+    const std::vector<SegmentChunkPair<T>>& shared_segment_chunk_pairs) {
+  if (jaccard_index >= jaccard_index_threshold) {
+    if (!_increases_attribute_vector_width(shared_dictionary_size, current_dictionary_size)) {
+      return std::none_of(shared_segment_chunk_pairs.cbegin(), shared_segment_chunk_pairs.cend(),
+                          [shared_dictionary_size](const SegmentChunkPair<T> segment_chunk_pair) {
+                            return _increases_attribute_vector_width(shared_dictionary_size,
+                                                                     segment_chunk_pair.first->dictionary()->size());
+                          });
+    }
+  }
+  return false;
+}
+
+template <typename T>
+void SharedDictionariesColumnProcessor<T>::_add_segment_chunk_pair(MergePlan& merge_plan,
```

Comment:
Am I wrong or does that seem like a method that could be implemented  (or is even better suited) as part of the `MergePlan` class itself.

Reply:
Fixed in https://github.com/hyrise/hyrise/pull/2394/commits/7f88317d0867f9bfd92688f5061daea73a39be35